### PR TITLE
Updates challenge creation controller API to properly handle JSON data in the request body

### PR DESCRIPTION
**Pull request by nucat176**

### CODE REVIEW

File: src/main/controllers/ChallengeController.kt [hunk 1]
```diff
@@ -17,8 +17,7 @@ class ChallengeController: DefaultController<Challenge>(), RestController<Challe
     override fun create(user: UserAccount?, requestData: RequestData): SOAResult<Challenge> {
         validateApiKey(user!!, requestData)
         return DaoService.execute {
-            val challengeNamespace = JsonHelper.parse<ChallengeNamespace>(requestData.body["challengeNamespace"]!! as String)
-
+            val challengeNamespace = JsonHelper.parse<ChallengeNamespace>(Klaxon().toJsonString(requestData.body["challengeNamespace"]))
```
Comment (lostcodingsomewhere) [inline src/main/controllers/ChallengeController.kt]: instead of initializing another Klaxon here, just us ethe one in JsonHelper

Comment (lostcodingsomewhere) [reply src/main/controllers/ChallengeController.kt]: what is in the body? its not a json string already?

Comment (nucat176) [reply src/main/controllers/ChallengeController.kt]: the logs indicate that its a JsonObject, not a String

Comment (lostcodingsomewhere) [reply src/main/controllers/ChallengeController.kt]: ya thats something u shouldnt change here -- change the test to pass a string

Comment (lostcodingsomewhere) [reply src/main/controllers/ChallengeController.kt]: actually it should be a Map -- if u look at RequestData how its populating body

**Pull request closed without merging**